src/main/java/org/mockito/internal/stubbing/answers/ThrowsException.java

```diff
@@ -29,6 +29,10 @@ public Object answer(InvocationOnMock invocation) throws Throwable {
             throw throwable;
         }
         Throwable t = throwable.fillInStackTrace();
+
+        if (t == null) {
+            throw throwable;
+        }
```

Comment:
Actually I'd rather make the check in `ConditionalStackTraceFilter.filter()`. And if throwable is null, just return.